src/main/resources/com/devonfw/ide/sonarqube/common/rules/devon4j/E7_java.html

```diff
@@ -0,0 +1,52 @@
+<p>
+    This rule checks that all Use-Case implementation methods are annotated with a security constraint.
+    Valid security constraint annotations are the following from the javax.annotation.security package:
+    <ul type='disc'>
+        <li>@DenyAll</li>
+        <li>@PermitAll</li>
+        <li>@RolesAllowed</li>
+    </ul>
+</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+    public class TestInterfaceImpl implements TestInterface {
+
+        @Override                                 
+        private boolean testBoolMethod() {
+            return true;
+        }
+    }
+</pre>
+<h2>Compliant Solutions</h2>
+<pre>
+    public class TestClassImpl implements TestClass {
+
+        @Override
+        @PermitAll
+        public void testMethod() {
+
+        }
+    }
+</pre>
+<br>
+<pre>
+    public class TestClassImpl implements TestClass {
+
+        @Override
+        @DenyAll
+        public void testMethod() {
+
+        }  
+    }
+</pre>
```

Comment:
Technically correct. However, these examples are more or less irrelevant.
In order not to leat people come to the wrong ideas, I would rather explain below as text that `@PermitAll` and `@DenyAll` are also accepted though they should only be used for edge-cases.